src/control/drpc/modules.go

```diff
@@ -49,59 +53,85 @@ const (
 	MethodRequestCredentials = C.DRPC_METHOD_SEC_AGENT_REQUEST_CREDS
 )
 
+// MgmtMethod is a type alias for a drpc mgmt method.
+type MgmtMethod int32
+
+func (mm MgmtMethod) String() string {
+	return map[MgmtMethod]string{
+		MethodPrepShutdown: "prep shutdown",
+		MethodPingRank:     "ping",
+		MethodSetRank:      "set rank",
+		MethodSetUp:        "setup MS",
+	}[mm]
+}
+
+// Module returns the module that the method belongs to.
+func (mm MgmtMethod) Module() int32 {
+	return ModuleMgmt
+}
+
+// TargetState returns the system member state that should be set on successful
+// completion of the method action.
+func (mm MgmtMethod) TargetState() system.MemberState {
+	return map[MgmtMethod]system.MemberState{
+		MethodPrepShutdown: system.MemberStateStopping,
+		MethodPingRank:     system.MemberStateReady,
+	}[mm]
+}
```

Comment:
This is an interesting idea, but I don't think it belongs here in the drpc package. Probably better to define the transition map in the system package, right?